File: BSharp/EntityModel/IfrsNote.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BSharp.EntityModel
{
    [StrongEntity]
    public class IfrsNote : IfrsConcept
    {
        [AlwaysAccessible]
        public short? Level { get; set; }

        [AlwaysAccessible]
        public string ParentId { get; set; }

        [AlwaysAccessible]
        public int? ActiveChildCount { get; set; }

        [AlwaysAccessible]
        public int? ChildCount { get; set; }

        [Display(Name = "IfrsNotes_IsAggregate")]
        public bool? IsAggregate { get; set; }

        [Display(Name = "IfrsNotes_ForDebit")]
        public bool? ForDebit { get; set; }

        [Display(Name = "IfrsNotes_ForCredit")]
        public bool? ForCredit { get; set; }

        // For Query

        [AlwaysAccessible]
        public HierarchyId Node { get; set; }

        [AlwaysAccessible]
        public HierarchyId ParentNode { get; set; }

        [Display(Name = "TreeParent")]
        [ForeignKey(nameof(ParentId))]
        public IfrsNote Parent { get; set; }

        [Display(Name = "CreatedBy")]
        [ForeignKey(nameof(CreatedById))]
        public User CreatedBy { get; set; }

        [Display(Name = "ModifiedBy")]
        [ForeignKey(nameof(ModifiedById))]
        public User ModifiedBy { get; set; }
    }

}

File: Tellma/ClientApp/src/app/data/entities/ifrs-note.ts
// tslint:disable:variable-name
import { SettingsForClient } from '../dto/settings-for-client';
import { EntityDescriptor } from './base/metadata';
import { WorkspaceService } from '../workspace.service';
import { TranslateService } from '@ngx-translate/core';
import { IfrsConcept, metadata_IfrsConceptInner } from './ifrs-concept';

export interface IfrsNote extends IfrsConcept {
  Node?: string;
  Level?: number;
  ParentNode?: string;
  ChildCount?: number;
  IsAggregate?: boolean;
  ForDebit?: boolean;
  ForCredit?: boolean;
}

// Choice list
export const IfrsConcept_IfrsType = {
  Amendment: 'IfrsConcept_Amendment',
  Extension: 'IfrsConcept_Extension',
  Regulatory: 'IfrsConcept_Regulatory'
};

let _settings: SettingsForClient;
let _cache: EntityDescriptor;

export function metadata_IfrsNote(wss: WorkspaceService, trx: TranslateService, definitionId: string): EntityDescriptor {
  const ws = wss.currentTenant;
  // Some global values affect the result, we check here if they have changed, otherwise we return the cached result
  if (ws.settings !== _settings) {
    _settings = ws.settings;
    _cache = metadata_IfrsConceptInner(wss, trx, definitionId);
    _cache.collection = 'IfrsNote';
    _cache.apiEndpoint = 'ifrs-notes';
    _cache.screenUrl = 'ifrs-notes';
    _cache.properties.IsAggregate = { control: 'boolean', label: () => trx.instant('IfrsNotes_IsAggregate') };
    _cache.properties.ForDebit = { control: 'boolean', label: () => trx.instant('IfrsNotes_ForDebit') };
    _cache.properties.ForCredit = { control: 'boolean', label: () => trx.instant('IfrsNotes_ForCredit') };

    // tree stuff
    _cache.properties.ChildCount = {
      control: 'number', label: () => trx.instant('TreeChildCount'), minDecimalPlaces: 0,
      maxDecimalPlaces: 0, alignment: 'right'
    };
    _cache.properties.ActiveChildCount = {
      control: 'number', label: () => trx.instant('TreeActiveChildCount'),
      minDecimalPlaces: 0, maxDecimalPlaces: 0, alignment: 'right'
    };
    _cache.properties.Level = {
      control: 'number', label: () => trx.instant('TreeLevel'),
      minDecimalPlaces: 0, maxDecimalPlaces: 0, alignment: 'right'
    };
    _cache.properties.Parent = {
      control: 'navigation', label: () => trx.instant('TreeParent'),
      type: 'ProductCategory', foreignKeyName: 'ParentId'
    };
  }

  return _cache;
}
